src/similarity.rs

```diff
@@ -120,7 +150,12 @@ where
         let mut embedding = (&embedding2.as_view() - &embedding1.as_view()) + embedding3.as_view();
         l2_normalize(embedding.view_mut());
 
-        let skip = [word1, word2, word3].iter().cloned().collect();
+        let skip = [word1, word2, word3]
+            .iter()
+            .zip(mask.iter())
+            .filter(|(_, &include)| !include)
```

Comment:
See above, I would expect masking to exclude the words (*masking* in the *conceal* sense). Perhaps mask is not the correct verb to use, maybe *exclude*?